packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryUtils.java

```diff
@@ -215,6 +213,27 @@ static Integer convertMediaType(String mediaType) throws IllegalArgumentExceptio
     return MEDIA_TYPES.get(mediaType);
   }
 
+  static int[] getSizeFromCursor(Cursor cursor, int mediaType, int localUriIndex){
+    final int orientationIndex = cursor.getColumnIndex(Media.ORIENTATION);
+    final int widthIndex = cursor.getColumnIndex(Media.WIDTH);
+    final int heightIndex = cursor.getColumnIndex(Media.HEIGHT);
+
+    int[] size;
+    // If image doesn't have the required information, we can get them from Bitmap.Options
+    if (cursor.getType(widthIndex) == Cursor.FIELD_TYPE_NULL &&
+        cursor.getType(heightIndex) == Cursor.FIELD_TYPE_NULL &&
```

Comment:
I believe if any of `cursor.getType({widthIndex, heightIndex})` equals `Cursor.FIELD_TYPE_NULL` we should check for actual size using this different flow. 
```suggestion
if ((cursor.getType(widthIndex) == Cursor.FIELD_TYPE_NULL ||
    cursor.getType(heightIndex) == Cursor.FIELD_TYPE_NULL) &&
    mediaType == Files.FileColumns.MEDIA_TYPE_IMAGE)
```